### PR TITLE
OBSDOCS-218: Logging 5.7.2 Release Notes

### PR DESCRIPTION
**Pull request by libander**
Version(s): 4.11+

Issue: [OBSDOCS-218](https://issues.redhat.com//browse/OBSDOCS-218) 

Link to docs preview: https://drive.google.com/file/d/1tPmyKXOGQOHwka8nRDbxzm6YqXtvJlXj/view?usp=drive_link

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

**Review by jcantrill — COMMENTED** [review]
/lgtm

**Review by cahartma — COMMENTED** [review]
/lgtm

**Comment by kabirbhartiRH** [comment]
The following bugs do not have release notes/no-rn label
https://issues.redhat.com/browse/LOG-4163
https://issues.redhat.com/browse/LOG-3314
https://issues.redhat.com/browse/LOG-4011
https://issues.redhat.com/browse/LOG-4098
https://issues.redhat.com/browse/LOG-4151
https://issues.redhat.com/browse/LOG-4218
https://issues.redhat.com/browse/LOG-4219
https://issues.redhat.com/browse/LOG-4220
https://issues.redhat.com/browse/LOG-4221

**Comment by bburt-rh** [comment]
lgtm

File: modules/logging-rn-5.7.2.adoc [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+* Before this change when configuring log forwarding to write to a Kafka output topic which matched a key in the payload, logs dropped due to an error.  With this update, Fluentd's buffer name has been prefixed with an underscore, resolving the issue.(link:https://issues.redhat.com/browse/LOG-4027[LOG-4027])
+
+* Before this update, the LokiStack gateway returned label values for namespaces without applying the access rights of a user. With this update, the LokiStack-gateway applies permissions to label value requests, resolving the issue. (link:https://issues.redhat.com/browse/LOG-4049[LOG-4049])
+
+* Before this update, the ClusterLogging Operator API required a certificate to be provided by a secret when the `tls.insecureSkipVerify` option was set to `true`. With this update, the ClusterLogging Operator API will no longer require a certificate to be provided by a secret in such cases. The following configuration has been added to the Operator's CR:
+[source,yaml]
```
Comment (rkratky) [inline modules/logging-rn-5.7.2.adoc:31]:
```suggestion
+
[source,yaml]
```

File: modules/logging-rn-5.7.2.adoc [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+----
+tls.verify_certificate = false
+tls.verify_hostname = false
+----
+(link:https://issues.redhat.com/browse/LOG-3445[LOG-3445])
```
Comment (rkratky) [inline modules/logging-rn-5.7.2.adoc:36]:
```suggestion
+
(link:https://issues.redhat.com/browse/LOG-3445[LOG-3445])
```

**Comment by libander** [comment]
Updated preview: https://drive.google.com/file/d/138xzss12O-x2ksRbY_Uiclk18vhvVC7F/view?usp=sharing

File: modules/logging-rn-5.7.2.adoc [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+// cluster-logging-release-notes.adoc
+:_content-type: REFERENCE
+[id="cluster-logging-release-notes-5-7-2_{context}"]
+= Logging 5.7.2
+This release includes link:https://access.redhat.com/errata/RHSA-2023:3495[OpenShift Logging Bug Fix Release 5.7.2].
+
+[id="openshift-logging-5-7-2-bug-fixes_{context}"]
+== Bug fixes
+* Before this update, it was not possible to delete the `openshift-logging` namespace directly due to the presence of a pending finalizer. With this update, the finalizer is no longer utilized, enabling direct deletion of the namespace. (link:https://issues.redhat.com/browse/LOG-3316[LOG-3316])
+
```
Comment (rh-tokeefe) [inline modules/logging-rn-5.7.2.adoc:11]: * Before this update, the `run.sh` script would display an incorrect `chunk_limit_size` value if it was changed according to the {product-title} documentation.

File: modules/logging-rn-5.7.2.adoc [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+* Before this update, Cluster Logging Operator encountered an Unsupported Media Type exception when trying to send logs to DataDog via the Fluentd HTTP Plugin. With this update, users can seamlessly assign the content type for log forwarding by configuring the HTTP header Content-Type. The value provided is automatically assigned to the `content_type` parameter within the plugin, ensuring successful log transmission. (link:https://issues.redhat.com/browse/LOG-3784[LOG-3784])
+
+* Before this update, when the `detectMultilineErrors` field was set to `true` in the `ClusterLogForwarder` Custom Resource (CR), PHP multi-line errors were recorded as separate log entries, causing the stack trace to be split across multiple messages. With this update, multi-line error detection for PHP is enabled, ensuring that the entire stack trace is included in a single log message. (link:https://issues.redhat.com/browse/LOG-3878[LOG-3878])
+
```
Comment (rh-tokeefe) [inline modules/logging-rn-5.7.2.adoc:19]: Should ClusterLogForwarder be tagged `ClusterLogForwarder`?

File: modules/logging-rn-5.7.2.adoc [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+[source,yaml]
+----
+tls.verify_certificate = false
+tls.verify_hostname = false
+----
++
+(link:https://issues.redhat.com/browse/LOG-3445[LOG-3445])
+
+* Before this update, the LokiStack route configuration caused queries running longer than 30 seconds to timeout. With this update, the LokiStack global and per-tenant `queryTimeout` settings affects the route timeout settings, resolving the issue. (link:https://issues.redhat.com/browse/LOG-4052[LOG-4052])
+
```
Comment (rh-tokeefe) [inline modules/logging-rn-5.7.2.adoc:41]: "prefer" is written twice, delete one instance.

File: modules/logging-rn-5.7.2.adoc [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+* Before this update, the Vector log collector did not generate TLS configuration for forwarding logs to multiple Kafka brokers if the broker URLs were not specified in the output. With this update, TLS configuration is generated appropriately for multiple brokers. (link:https://issues.redhat.com/browse/LOG-4163[LOG-4163])
+
+* Before this update, the option to enable passphrase for log forwarding to Kafka was unavailable. This limitation presented a security risk as it could potentially expose sensitive information. With this update, users now have a seamless option to enable passphrase for log forwarding to Kafka. (link:https://issues.redhat.com/browse/LOG-3314[LOG-3314])
+
+* Before this update, Vector log collector did not honor the `tlsSecurityProfile `settings for outgoing TLS connections. After this update, Vector handles TLS connection settings appropriately. (link:https://issues.redhat.com/browse/LOG-4011[LOG-4011])
+
```
Comment (rh-tokeefe) [inline modules/logging-rn-5.7.2.adoc:49]: I think you can delete "now" from the clause "With this update, metrics now contain..."

File: modules/logging-rn-5.7.2.adoc [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+* Before this update, when `follow_inodes` was set to `true`. the Fluentd collector could crash on file rotation. With this update, the `follow_inodes` setting does not crash the collector. (link:https://issues.redhat.com/browse/LOG-4151[LOG-4151])
+
+* Before this update the Fluentd collector could incorrectly close files that it should be watching because of the way those files were tracked. With this update, the tracking parameters have been corrected. (link:https://issues.redhat.com/browse/LOG-4149[LOG-4149])
+
+* Before this update, forwarding logs with the Vector collector and naming a pipeline in the `ClusterLogForwarder` instance `audit`, `application` or `infrastructure` resulted in collector pods staying in the `CrashLoopBackOff` state with the following error in the collector log: "ERROR vector::cli: Configuration error. error=redefinition of table `transforms.audit` for key `transforms.audit`." After this update, pipeline names no longer clash with reserved input names and pipelines can be named 'audit', 'application' or 'infrastructure'. (link:https://issues.redhat.com/browse/LOG-4218[LOG-4218])
+
```
Comment (rh-tokeefe) [inline modules/logging-rn-5.7.2.adoc:57]: I think this is mostly personal preference, but how about rephrasing " With this update, these fields are no longer added to journal logs" to " With this update, journal logs do not contain these fields."

File: modules/logging-rn-5.7.2.adoc (hunk 4 above)
Comment (libander) [reply modules/logging-rn-5.7.2.adoc:19]: ...yes.

File: modules/logging-rn-5.7.2.adoc (hunk 6 above)
Comment (libander) [reply modules/logging-rn-5.7.2.adoc:49]: Not a single one of your comments seems to be associated with the appropriate lines you're referring to, FYI.

**Comment by libander** [comment]
https://drive.google.com/file/d/1s5WXHNanVVagUS6PO9fVVN4aKp5iKAws/view?usp=sharing

File: modules/logging-rn-5.7.2.adoc [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+* Before this update, when `follow_inodes` was set to `true`. the Fluentd collector could crash on file rotation. With this update, the `follow_inodes` setting does not crash the collector. (link:https://issues.redhat.com/browse/LOG-4151[LOG-4151])
+
+* Before this update the Fluentd collector could incorrectly close files that it should be watching because of the way those files were tracked. With this update, the tracking parameters have been corrected. (link:https://issues.redhat.com/browse/LOG-4149[LOG-4149])
+
+* Before this update, forwarding logs with the Vector collector and naming a pipeline in the `ClusterLogForwarder` instance `audit`, `application` or `infrastructure` resulted in collector pods staying in the `CrashLoopBackOff` state with the following error in the collector log: "ERROR vector::cli: Configuration error. error=redefinition of table `transforms.audit` for key `transforms.audit`." After this update, pipeline names no longer clash with reserved input names and pipelines can be named 'audit', 'application' or 'infrastructure'. (link:https://issues.redhat.com/browse/LOG-4218[LOG-4218])
```
Comment (abrennan89) [inline modules/logging-rn-5.7.2.adoc:56]:
```suggestion
* Before this update, forwarding logs with the Vector collector and naming a pipeline in the `ClusterLogForwarder` instance named `audit`, `application` or `infrastructure` resulted in collector pods staying in the `CrashLoopBackOff` state with the following error in the collector log:
+
[source,text]
----
ERROR vector::cli: Configuration error. error=redefinition of table transforms.audit for key transforms.audit
----
+
After this update, pipeline names no longer clash with reserved input names, and pipelines can be named 'audit', 'application' or 'infrastructure'. (link:https://issues.redhat.com/browse/LOG-4218[LOG-4218])
```

think "named" or similar word was missing here; also, longer system errors should be presented in a code block, per https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#system-messages

File: modules/logging-rn-5.7.2.adoc [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+
+* Before this update, when forwarding logs to a syslog destination with the vector collector and setting the `addLogSource` flag to `true`, extra empty fields were added to the forwarded messages: `namespace_name=`, `container_name=`, `pod_name=`. With this update, these fields are no longer added to journal logs. (link:https://issues.redhat.com/browse/[LOG-4219])
```
Comment (abrennan89) [inline modules/logging-rn-5.7.2.adoc:58]:
```suggestion
* Before this update, when forwarding logs to a syslog destination with the Vector collector and setting the `addLogSource` flag to `true`, the following extra empty fields were added to the forwarded messages: `namespace_name=`, `container_name=`, and `pod_name=`. With this update, these fields are no longer added to journal logs. (link:https://issues.redhat.com/browse/[LOG-4219])
```

**Review by abrennan89 — COMMENTED** [review]
Some errors that need to be fixed before this can be merged. Thanks.

File: modules/logging-rn-5.7.2.adoc [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+
+* Before this update, the {product-title} web console's logging view plugin did not allow for custom node placement or tolerations. This update adds the ability to define node placement and tolerations for the logging view plugin. (link:https://issues.redhat.com/browse/LOG-3749[LOG-3749])
+
+* Before this update, Cluster Logging Operator encountered an Unsupported Media Type exception when trying to send logs to DataDog via the Fluentd HTTP Plugin. With this update, users can seamlessly assign the content type for log forwarding by configuring the HTTP header Content-Type. The value provided is automatically assigned to the `content_type` parameter within the plugin, ensuring successful log transmission. (link:https://issues.redhat.com/browse/LOG-3784[LOG-3784])
```
Comment (abrennan89) [inline modules/logging-rn-5.7.2.adoc:16]:
```suggestion
* Before this update, the Cluster Logging Operator encountered an Unsupported Media Type exception when trying to send logs to DataDog via the Fluentd HTTP Plugin. With this update, users can seamlessly assign the content type for log forwarding by configuring the HTTP header Content-Type. The value provided is automatically assigned to the `content_type` parameter within the plugin, ensuring successful log transmission. (link:https://issues.redhat.com/browse/LOG-3784[LOG-3784])
```

File: modules/logging-rn-5.7.2.adoc [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+
+* Before this update, when the `detectMultilineErrors` field was set to `true` in the `ClusterLogForwarder` custom resource (CR), PHP multi-line errors were recorded as separate log entries, causing the stack trace to be split across multiple messages. With this update, multi-line error detection for PHP is enabled, ensuring that the entire stack trace is included in a single log message. (link:https://issues.redhat.com/browse/LOG-3878[LOG-3878])
+
+* Before this update, `ClusterLogForwarder` pipelines containing a space in their name caused the Vector collector pods to continuously crash. With this update, all spaces, dashes (-), and dots (.) in pipeline names are replaced with underscores (_). (link:https://issues.redhat.com/browse/LOG-3945[LOG-3945])
+
+* Before this update, the `log_forwarder_output` metric did not include the http parameter. This update adds the missing parameter to the metric. (link:https://issues.redhat.com/browse/LOG-3997[LOG-3997])
```
Comment (abrennan89) [inline modules/logging-rn-5.7.2.adoc:22]:
```suggestion
* Before this update, the `log_forwarder_output` metric did not include the `http` parameter. This update adds the missing parameter to the metric. (link:https://issues.redhat.com/browse/LOG-3997[LOG-3997])
```

File: modules/logging-rn-5.7.2.adoc [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+
+* Before this update, Fluentd did not identify some multi-line JavaScript client exceptions when they ended with a colon. With this update, the Fluentd buffer name is prefixed with an underscore, resolving the issue. (link:https://issues.redhat.com/browse/LOG-4019[LOG-4019])
+
+* Before this update, when configuring log forwarding to write to a Kafka output topic which matched a key in the payload, logs dropped due to an error.  With this update, Fluentd's buffer name has been prefixed with an underscore, resolving the issue.(link:https://issues.redhat.com/browse/LOG-4027[LOG-4027])
+
+* Before this update, the LokiStack gateway returned label values for namespaces without applying the access rights of a user. With this update, the LokiStack-gateway applies permissions to label value requests, resolving the issue. (link:https://issues.redhat.com/browse/LOG-4049[LOG-4049])
```
Comment (abrennan89) [inline modules/logging-rn-5.7.2.adoc:28]:
```suggestion
* Before this update, the LokiStack gateway returned label values for namespaces without applying the access rights of a user. With this update, the LokiStack gateway applies permissions to label value requests, resolving the issue. (link:https://issues.redhat.com/browse/LOG-4049[LOG-4049])
```

File: modules/logging-rn-5.7.2.adoc [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+----
+tls.verify_certificate = false
+tls.verify_hostname = false
+----
++
+(link:https://issues.redhat.com/browse/LOG-3445[LOG-3445])
+
+* Before this update, the LokiStack route configuration caused queries running longer than 30 seconds to timeout. With this update, the LokiStack global and per-tenant `queryTimeout` settings affect the route timeout settings, resolving the issue. (link:https://issues.redhat.com/browse/LOG-4052[LOG-4052])
+
+* Before this update, a prior fix to remove defaulting of the `collection.type` resulted in the Operator no longer honoring the deprecated specs for resource, node selections, and tolerations.  This update modifies the Operator behavior to always prefer the `collection.logs` spec over those of `collection`.  Note this varies from previous behavior that allowed using both the preferred fields and deprecated fields but would ignore the deprecated fields when `collection.type` was populated. (link:https://issues.redhat.com/browse/LOG-4185[LOG-4185])
```
Comment (abrennan89) [inline modules/logging-rn-5.7.2.adoc:42]:
```suggestion
* Before this update, a prior fix to remove defaulting of the `collection.type` resulted in the Operator no longer honoring the deprecated specs for resource, node selections, and tolerations.  This update modifies the Operator behavior to always prefer the `collection.logs` spec over those of `collection`. This varies from previous behavior that allowed using both the preferred fields and deprecated fields but would ignore the deprecated fields when `collection.type` was populated. (link:https://issues.redhat.com/browse/LOG-4185[LOG-4185])
```

File: modules/logging-rn-5.7.2.adoc [hunk 14]
```diff
@@ -0,0 +1,122 @@
...
+
+* Before this update, not all available output types were included in the `log_forwarder_output_info` metrics. With this update, metrics contain Splunk and Google Cloud Logging data which was missing previously. (link:https://issues.redhat.com/browse/LOG-4098[LOG-4098])
+
+* Before this update, when `follow_inodes` was set to `true`, the Fluentd collector could crash on file rotation. With this update, the `follow_inodes` setting does not crash the collector. (link:https://issues.redhat.com/browse/LOG-4151[LOG-4151])
+
+* Before this update, the Fluentd collector could incorrectly close files that should be watched because of how those files were tracked. With this update, the tracking parameters have been corrected.  (link:https://issues.redhat.com/browse/LOG-4149[LOG-4149])
+
+* Before this update, forwarding logs with the Vector collector and naming a pipeline in the `ClusterLogForwarder` instance named `audit`, `application` or `infrastructure` resulted in collector pods staying in the `CrashLoopBackOff` state with the following error in the collector log:
```
Comment (abrennan89) [inline modules/logging-rn-5.7.2.adoc:56]:
```suggestion
* Before this update, forwarding logs with the Vector collector and naming a pipeline in the `ClusterLogForwarder` instance `audit`, `application` or `infrastructure` resulted in collector pods staying in the `CrashLoopBackOff` state with the following error in the collector log:
```
I think this one was actually my bad, I must've missed that it says "naming" earlier in the sentence.

File: modules/logging-rn-5.7.2.adoc [hunk 15]
```diff
@@ -0,0 +1,122 @@
...
++
+[source,text]
+----
+ERROR vector::cli: Configuration error. error=redefinition of table transforms.audit for key transforms.audit
+----
++
+After this update, pipeline names no longer clash with reserved input names, and pipelines can be named 'audit', 'application' or 'infrastructure'. (link:https://issues.redhat.com/browse/LOG-4218[LOG-4218])
```
Comment (abrennan89) [inline modules/logging-rn-5.7.2.adoc:63]:
```suggestion
After this update, pipeline names no longer clash with reserved input names, and pipelines can be named `audit`, `application` or `infrastructure`. (link:https://issues.redhat.com/browse/LOG-4218[LOG-4218])
```

**Review by abrennan89 — COMMENTED** [review]
few more minor things

**Review by abrennan89 — APPROVED** [review]
/lgtm
🥳

**Pull request merged**